simplifier/world/nodes.py

```diff
@@ -114,6 +114,15 @@ def simplify(self):
         for node in list(self.world.iter_postorder(self)):
             if isinstance(node, BitwiseNegate):
                 node.dissolve_negation()
+        from simplifier.range_simplifier import RangeSimplifier
+
+        for node in list(self.world.iter_postorder(self)):
+            if isinstance(node, Operation):
+                RangeSimplifier.simplify(node)
+                from simplifier.operations.interface import AssociativeOperation, CommutativeOperation
```

Comment:
could we have all the imports on top of the function?

Reply:
Unfortunately this leads to circular import problems :(